File: 100-Days-Series/Day 43/RotateMatrix90d.js

// Arrow Function to rotate a matrix by 90 degrees clockwise
const getRotate=(matrix)=>{
    let n=matrix.length;
    for(let i=0;i<n;i++){
        for(let j=i+1;j<n;j++){
            temp=matrix[i][j];
            matrix[i][j]=matrix[j][i];
            matrix[j][i]=temp;
        }
    }
    for(let i=0;i<n;i++){
        let left=0;
        let right=n-1;
        while(left<right){
            temp=matrix[i][left];
            matrix[i][left]=matrix[i][right];
            matrix[i][right]=temp;
            left++;
            right--;
        }
    }
}
// Function to print matrix
const getPrintMatrix=(matrix)=>{
    for(let i=0;i<matrix.length;i++){
        for(let j=0;j<matrix.length;j++){
            console.log("Matrix["+i+"]["+j+"] = "+matrix[i][j]);
        }
    }
}
// Let's Start here
matrix=[
    [1,2,3],
    [4,5,6],
    [7,8,9]
];
console.log("Original Matrix :");
getPrintMatrix(matrix);
console.log("Rotated Matrix : ");
getRotate(matrix);
getPrintMatrix(matrix)

File: 100-Days-Series/Day 43/RotateMatrix90d.c
#include<stdio.h>
// Function to Rotate a matrix by 90 degrees clockwise.
void getRotate(int n,int matrix[n][n]){
    // Step 1: Transpose the matrix
    for(int i=0;i<n;i++){
        for(int j=i+1;j<n;j++){
            int temp=matrix[i][j];
            matrix[i][j]=matrix[j][i];
            matrix[j][i]=temp;
        }
    }
    // Step 2: Reverse each row
    for(int i=0;i<n;i++){
        int left=0,right=n-1;
        while(left<right){
            int temp=matrix[i][left];
            matrix[i][left]=matrix[i][right];
            matrix[i][right]=temp;
            left++;
            right--;
        }
    }
}
// function for print matrix
void getPrintMatrix(int size,int matrix[size][size]){
    for(int i=0;i<size;i++){
        for(int j=0;j<size;j++){
            printf("%d ",matrix[i][j]);
        }
        printf("\n");
    }
}
// Let's Start here
void main(){
    int matrix[3][3]={
        {1,2,3},
        {4,5,6},
        {7,8,9}
    };
    int n=3;
    printf("\nOriginal Matrix :\n");
    getPrintMatrix(n,matrix);
    printf("\nRotated Matrix :\n");
    getRotate(n,matrix);
    getPrintMatrix(n,matrix);
}
